src/XMakeTasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -1558,9 +1558,9 @@ Copyright (C) Microsoft Corporation. All rights reserved.
         Projects="%(_MSBuildProjectReferenceExistent.Identity)"
         Targets="GetTargetFrameworkProperties"
         BuildInParallel="$(BuildInParallel)"
-        Properties="%(_MSBuildProjectReferenceExistent.SetConfiguration); %(_MSBuildProjectReferenceExistent.SetPlatform); ReferringTargetFramework=$(ReferringTargetFrameworkForProjectReferences)"
+        Properties="%(_MSBuildProjectReferenceExistent.SetConfiguration); %(_MSBuildProjectReferenceExistent.SetPlatform); ReferringTargetFramework=$(ReferringTargetFrameworkForProjectReferences); ReferringRuntimeIdentifier=$(RuntimeIdentifier)"
         ContinueOnError="!$(BuildingProject)"
-        RemoveProperties="%(_MSBuildProjectReferenceExistent.GlobalPropertiesToRemove);TargetFramework"
+        RemoveProperties="%(_MSBuildProjectReferenceExistent.GlobalPropertiesToRemove);TargetFramework;RuntimeIdentifier"
```

Comment:
I think this new dimension will reintroduce build races between sln traversal and P2P traversal. Sln traversal will build all projects without /p:RuntimeIdentifier, but P2P traversal will always pass /p:RuntimeIdentifier. This will cause MSBuild to think it can do both builds in parallel but they both output to the same directory by default.

I think we need two things:

1. RID (if non-empty) must be appended to build output path by default
2. We must add RuntimeIdentifier to GlobalPropertiesToRemove if it comes back empty

(1) is something we just shied away from as being too risky :(